packages/dd-trace/src/profiling/profilers/events.js

```diff
@@ -199,19 +194,22 @@ class EventsProfiler {
     })()
 
     const decorators = {}
-    for (const [eventType, DecoratorCtor] of Object.entries(decoratorTypes)) {
-      const decorator = new DecoratorCtor(stringTable)
-      decorator.eventTypeLabel = labelFromStrStr(stringTable, 'event', eventType)
-      decorators[eventType] = decorator
-    }
     const timestampLabelKey = stringTable.dedup(END_TIMESTAMP_LABEL)
 
     const dateOffset = BigInt(Math.round(performance.timeOrigin * MS_TO_NS))
     const lateEntries = []
     const perfEndDate = endDate.getTime() - performance.timeOrigin
     const samples = this.entries.map((item) => {
-      const decorator = decorators[item.entryType]
+      let decorator = decorators[item.entryType]
       if (!decorator) {
+        const eventType = item.entryType
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🤓 nitpick:** ‏ `const eventType = item.entryType` could be moved up before `let decorator = decorators[item.entryType]`

Reply:
While I'm usually all for common subexpression elimination, the idea here was that in majority of cases, the expression is evaluated only once in `decorators[item.entryType]`, so a store/load to/from a local variable is unnecessary. At the end of the day, you're right and I should ween myself off of these kinds of microoptimizations; readability is more important :-)

Reply:
I had not thought about the optimization point of view, interesting take that sometimes CSE could be detrimental to performance.